components/tidb_query_datatype/src/codec/mysql/time/mod.rs

```diff
@@ -1343,6 +1343,13 @@ impl Time {
         Time::try_from_chrono_datetime(ctx, time, time_type, duration.fsp() as i8)
     }
 
+    pub fn from_local_time(ctx: &mut EvalContext, time_type: TimeType, fsp: i8) -> Result<Time> {
```

Comment:
Why would TiDB push down now calculation to TiKV? It's meaningless as time on different servers are different, so the result is unpredictable.

Reply:
This question is discussed by Yu Dong, Eason ;  Simple conclusion is : sysdate is different with now(),  it can be push down as default:   see detail in [design doc](https://docs.google.com/document/d/10_9FP6M0mBqOhBJong8JJcbzkoOHaVc1ToVgntWT7VE/edit?n=Now_sysdate).

Reply:
pushing `sysdate` down is a required feature and confirmed by discussion from PM, the inconsistent result on different tikv-server is acceptable. And `sysdate` can also be pushed down to tiflash already.

Reply:
Given the natural that `sysdate` is  nondeterministic, it looks good to me. https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_sysdate